tools/tertiary-analysis/scanpy/scanpy-run-umap.xml

```diff
@@ -8,8 +8,16 @@
   <command detect_errors="exit_code"><![CDATA[
 ln -s '${input_obj_file}' input.h5 &&
 PYTHONIOENCODING=utf-8 scanpy-run-umap
-    --neighbors-key '${use_graph}'
+#set neighbors_key = $use_graph.replace('INPUT_OBJ', $input_obj_file.__getattr__('name'))
```

Comment:
for my understanding, could you clarify what you are trying to do here? Thanks

Reply:
I'm allowing the user to specify INPUT_OBJ as a wild card, to be used in specifying the slot to store the UMAP results in. It's a consequence of parameterising UMAP by n_neighbors, which is a a parameter of the upstream tool.  It's analagous to how we already derive the output file name for the UMAP embeddings.

I'm not proud of this- so if you have a better way I'd be grateful.